torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -87,6 +88,26 @@ namespace {
 
 constexpr int64_t kSeedInfoId = -127389;
 
+// Traits related to the return type of the lambda function that wraps the
+// actual implementation inside PythonScope.
+template <class T>
+struct LambdaReturnTypeTrait {
```

Comment:
The name should make its semantics obvious. Since this is used to unwrap Status/StatusOr, call it `StatusTypeTrait`?

Reply:
Semantically, I think `StatusTypeTrait` falls short, because its inner `Type` is also defined for non-status types:

- `StatusTypeTrait<T>::Type == T`
- `StatusTypeTrait<Status>::Type == void`
- `StatusTypeTrait<StatusOr<T>>::Type == T`

So, I named it `LambdaReturnTypeTrait` because it gives us the return type of the actual wrapping lambda, given the implementation function return type.

What do you think of: `WrapperLambdaReturn<ImplRetType>::type`?